src/Platforms/AbstractPlatform.php

```diff
@@ -396,6 +396,11 @@ abstract public function getBlobTypeDeclarationSQL(array $column);
      */
     abstract public function getName();
 
+    /**
+     * A flag that indicates whether the platform supports functional indexes.
+     */
+    abstract public function isSupportsFunctionalIndex(): bool;
```

Comment:
1. `isSupportsFunctionalIndex()` is an odd name. `supportsFunctionalIndex()` maybe?
2. Adding a new abstract method to the platform is a BC break for packages that add their own platforms. We should avoid this. Can we come up with a default implementation instead?
3. Does this method have to be public? As far as I can tell, it's only ever called by the platform itself.
4. Do we really need this method?

Reply:
1. Ok
2. Ok
3. Ok
4. I think yes, because various databases already support this feature and others may launch it in new versions.